### PR TITLE
Add support for symlinked folders in dashboard provisioning

### LINKED ISSUE #11958: Dashboard Provisioning does not resolve symlinks
Dashboard provisioning is not working for me when set to a symlink directory. Is this a known limitation? I've been searching around this github and the community site for a bit and haven't found any documentation about this.

### What Grafana version are you using?
We are testing out provisioning on the new 5.1.2 version

### What OS are you running grafana on?
Amazon Linux 2017.09

### What did you do?
Put json dashboards in a symlink "current release" directory (pointed to by the yaml file in the provisioning/dashboards directory). 

### What was the expected result?
Dashboards loaded

### What happened instead?
No Dashboards found
 
### More Details
In my provisioning yaml file, this is set: `path: /srv/www/dashboards/current`
This is a symlink directory: `current -> /srv/www/dashboards/releases/20180516223247`
When i change my yaml file to this: `path: /srv/www/dashboards/releases/20180516223247`, the dashboards load as expected.

There are no errors or messages related to dashboards or that specified directory in `/var/log/grafana/grafana.log`.

### PR DESCRIPTION
**Pull request by bergquist**
fixes #11958

### CODE REVIEW

File: pkg/services/provisioning/dashboards/file_reader.go [hunk 1]
```diff
@@ -47,15 +47,25 @@ func NewDashboardFileReader(cfg *DashboardsAsConfig, log log.Logger) (*fileReade
...
 
-	absPath, err := filepath.Abs(path)
+	copy := path
+	path, err := filepath.Abs(path)
 	if err != nil {
 		log.Error("Could not create absolute path ", "path", path)
-		absPath = path //if .Abs return an error we fallback to path
+	}
+
+	path, err = filepath.EvalSymlinks(path)
```
Comment (xlson) [inline pkg/services/provisioning/dashboards/file_reader.go]: Shouldn't we run `filepath.Abs(path)` again after evaluating any symbolic links given that EvalSymlinks gives a relative path if the linked path is relative?

Comment (xlson) [inline pkg/services/provisioning/dashboards/file_reader.go]: Shouldn't we run filepath.Abs(path) again after evaluating any symbolic links given that EvalSymlinks gives a relative path if the linked path is relative?

**Review by xlson — COMMENTED** [review]
Not sure about the absolute paths, looks good otherwise.

File: pkg/services/provisioning/dashboards/file_reader.go (hunk 1 above)
Comment (bergquist) [reply pkg/services/provisioning/dashboards/file_reader.go]: Good idea

**Pull request merged**